src/index.ts

```diff
@@ -70,25 +82,37 @@ export class PromiseQueue {
 			}
 
 			let timeout: ReturnType<typeof setTimeout> | undefined;
+			let enqueueTime: ReturnType<typeof process.hrtime>;
+			let serviceStartTime: ReturnType<typeof process.hrtime>;
 			const wrappedFn = async (e?: Error) => {
 				try {
+					this.metrics.emit('queueTime', durationSince(enqueueTime));
 					if (timeout) {
 						clearTimeout(timeout);
 					}
 					if (e) {
 						reject(e);
 						return;
 					}
+					this.metrics.emit('dequeue');
+					serviceStartTime = process.hrtime();
 					const result = await fn();
 					resolve(result);
 				} catch (e) {
 					reject(e);
+				} finally {
+					this.metrics.emit('serviceTime', durationSince(serviceStartTime));
+					this.metrics.emit('latency', durationSince(enqueueTime));
+					this.metrics.emit('completion');
 				}
 			};
 
 			this.queue.push(wrappedFn);
+			enqueueTime = process.hrtime();
```

Comment:
Can you declare/assign this in one go to a const please, which also removes the need to explicitly type it

Reply:
ack